src/forecast_solar/models.py

```
@@ -142,12 +144,10 @@ def peak_production_time(self, specific_date: date) -> datetime:
             (watt for date, watt in self.watts.items() if date.date() == specific_date),
             default=None,
         )
-        for (
-            timestamp,
-            watt,
-        ) in self.watts.items():
+        for timestamp, watt in self.watts.items():
             if watt == value:
                 return timestamp
+        return None
```

This will give some mypy issues in the future. I could set the return type `datetime | None`, but that also causes issues elsewhere who call the function 🤔

---

The `return None` is not necessary because we always expect a value, it is mainly Ruff who wants something to come here.

---

Then I'd raise eg RuntimeError.

---

I don't see how that would solve it or how that would be in the code.

---

Instead of returning Nome here, raise an exception. Then the return value won't be optional anymore.